android/app/src/main/java/org/fossasia/openevent/viewmodels/AboutFragmentViewModel.java

```diff
@@ -28,42 +28,36 @@
 
     public AboutFragmentViewModel() {
         realmRepo = RealmDataRepository.getDefaultInstance();
-        sessions = new MutableLiveData<>();
-        dateList = new MutableLiveData<>();
-        eventLiveData = new MutableLiveData<>();
-        subscribeToEvent();
-        subscribeToDateList();
-    }
-
-    private void subscribeToSessions(ArrayList<String> dates) {
-        LiveRealmData<Session> sessionLiveRealmData = RealmDataRepository.asLiveData(realmRepo.getBookMarkedSessions());
-        sessions = Transformations.map(sessionLiveRealmData, input -> getSessionsList(dates, input));
-    }
-
-    private void subscribeToDateList() {
-        RealmResults<EventDates> eventDates = realmRepo.getEventDatesSync();
-        ArrayList<String> dateListString = new ArrayList<>();
-        for (EventDates eventDate : eventDates) {
-            dateListString.add(eventDate.getDate());
-        }
-        dateList.setValue(dateListString);
-        subscribeToSessions(dateListString);
-    }
-
-    private void subscribeToEvent() {
-        LiveRealmDataObject<Event> liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getEvent());
-        eventLiveData = Transformations.map(liveRealmDataObject, input -> input);
     }
 
     public LiveData<Event> getEvent() {
+        if (eventLiveData == null) {
+            LiveRealmDataObject<Event> liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getEvent());
+            eventLiveData = Transformations.map(liveRealmDataObject, input -> input);
+        }
         return eventLiveData;
     }
 
     public LiveData<ArrayList<String>> getDateList() {
+        if (dateList == null) {
+            ArrayList<String> dateListString = new ArrayList<>();
+            if (dateList == null) {
```

Comment:
Why is the same condition nested?

Also, if using LiveData, use async method to get things and if using synchronous method, then don't return LiveData

Reply:
I have little confusion about the second part. Should I move the LiveRealmData code inside getBookmarkedSessions() and getEvent() to AboutFragmentViewModel() ?

Reply:
No, here you are using getEventDatesSync method which is a blocking call. So you don't need to return livedata. LiveData is for async calls and for data which changes

Reply:
I am having little trouble understanding how to achieve this. If I make getDateList method void, what changes should I make in the aboutFragment when this method is called ?

Reply:
`getDateList` should directly get the list of dates, because the function called inside is synchronous

```Java
public List<> getDateList() {
    ...
    return dateList;
}
```

Reply:
I think I did it correctly this time. Please review.
Thank you